pycbc/events/trigger_fits.py

```diff
@@ -91,17 +122,33 @@ def fit_above_thresh(distr, vals, thresh=None):
     vals = numpy.array(vals)
     if thresh is None:
         thresh = min(vals)
+        above_thresh = numpy.ones_like(vals, dtype=bool)
     else:
-        vals = vals[vals >= thresh]
-    alpha = fitalpha_dict[distr](vals, thresh)
-    return alpha, fitstd_dict[distr](vals, alpha)
+        above_thresh = vals >= thresh
+        if numpy.count_nonzero(above_thresh) == 0:
+            # Nothing is above threshold - warn and return -1
+            logging.warning("No values are above the threshold, %.2f, "
+                            "maximum is %.2f.", thresh, vals.max())
+            return -1., -1.
+
+        vals = vals[above_thresh]
+
+    # Set up the weights
+    if weights is not None:
+        weights = numpy.array(weights)
+        w = weights[above_thresh]
+    else:
+        w = numpy.ones_like(vals)
+
+    alpha = fitalpha_dict[distr](vals, thresh, w)
+    return alpha, fitstd_dict[distr](w, alpha)
 
 
 fitfn_dict = {
-    'exponential' : lambda x, alpha, t : alpha * numpy.exp(-alpha * (x - t)),
-    'rayleigh' : lambda x, alpha, t : (alpha * x * \
-                                       numpy.exp(-alpha * (x**2 - t**2) / 2.)),
-    'power' : lambda x, alpha, t : (alpha - 1.) * x**(-alpha) * t**(alpha - 1.)
+    'exponential' : lambda x, a, t : a * numpy.exp(-a * (x - t)),
```

Comment:
maybe worth putting a comment here to explain the variables - eg `x: the trigger stat value(s) at which to evaluate the function, a: slope parameter of the fit, t: lower threshold stat value`